plugins/zenoh-plugin-storage-manager/src/storages_mgt/mod.rs

```diff
@@ -46,28 +70,91 @@ pub(crate) async fn create_and_start_storage(
+        Ok(entries) => entries
+            .into_iter()
+            .map(|(stripped_key, ts)| {
+                (
+                    stripped_key.clone(),
+                    Event::new(stripped_key, ts, SampleKind::Put),
+                )
+            })
+            .collect::<HashMap<_, _>>(),
+        Err(e) => bail!("`get_all_entries` failed with: {e:?}"),
     };
 
+    let mut replication_log = None;
+    let mut latest_updates = HashMap::default();
+    if let Some(replica_config) = &config.replication {
+        if capability.history != History::Latest {
+            bail!(
+                "Replication was enabled for storage {name} but its history capability is not \
+                 supported: found < {:?} >, expected < {:?} >",
+                capability.history,
+                History::Latest
+            );
+        }
+        let mut log_latest = LogLatest::new(
+            config.key_expr.clone(),
+            config.strip_prefix.clone(),
+            replica_config.clone(),
+        );
+        log_latest.update(entries.drain().map(|(_, event)| event));
+
+        replication_log = Some(Arc::new(RwLock::new(log_latest)));
+    } else {
+        latest_updates = entries;
+    }
+
+    let latest_updates = Arc::new(RwLock::new(latest_updates));
+
     let storage = Arc::new(Mutex::new(storage));
-    tokio::task::spawn(async move {
-        StorageService::start(
-            zenoh_session,
-            config,
-            &name,
-            storage,
-            capability,
-            rx,
-            Arc::new(Mutex::new(latest_updates)),
-        )
-        .await;
-    });
+    let storage_service = StorageService::start(
+        zenoh_session.clone(),
+        config.clone(),
+        &name,
+        storage,
+        capability,
+        rx_storage,
+        CacheLatest::new(latest_updates.clone(), replication_log.clone()),
+    )
+    .await;
+
+    // Testing if the `replication_log` is set is equivalent to testing if the `replication` is
+    // set: the `replication_log` is only set when the latter is.
+    if let Some(replication_log) = replication_log {
+        let rx_replication = tx.subscribe();
+
+        // NOTE Although the function `ReplicationService::spawn_start` spawns its own tasks, we
+        //      still need to call it within a dedicated task because the Zenoh routing tables are
+        //      populated only after the plugins have been loaded.
+        //
+        //      If we don't wait for the routing tables to be populated the initial alignment
+        //      (i.e. querying any Storage on the network handling the same key expression), will
+        //      never work.
+        //
+        // TODO Do we really want to perform such an initial alignment? Because this query will
+        //      target any Storage that matches the same key expression, regardless of if they have
+        //      been configured to be replicated.
```

Comment:
If a storage is not configured as a replica, then it shouldn't receive digests correct ?
Will this not simply be ignored by the storage ?

Reply:
My comment stands: the code that follows makes a regular GET to a Storage that was declared on the same key expression. It does not leverage anything related to the Replication to filter out Storage that are not configured to be replicated.

I will open an issue.